app/Filters/TransactionsConditionFilter.php

```diff
@@ -51,19 +49,21 @@ public function handle(ConditionableContract $conditionable, Transaction ...$tra
                 // If there are no conditionals, apply the tag to everything.
                 return true;
             }
+
+            $returnValue = true;
             /** @var Condition $conditional */
             foreach ($conditions as $conditional) {
                 $condition = static::CONDITIONS[$conditional->getComparator()];
 
                 $condition = new $condition;
 
-                if ($condition($transaction, $conditional)) {
-                    // Only 1 of them needs to pass in order for the transaction to be "valid"
-                    return true;
+                if (!$condition($transaction, $conditional)) {
+                    $returnValue = false;
                 }
             }
 
-            return false;
+            // if any conditional fails, then we'll need to return false for this array filter so it's filtered out.
+            return $returnValue;
```

Comment:
OMFG. This logic was a pain for some reason. I guess it just didn't click until now... 🙃 Welp. It's all fixedup..